src/tfs/tlog.c

```diff
@@ -202,31 +440,89 @@ closure_function(2, 1, void, log_read_complete,
     tl->dictionary = newdict;
 #endif
 
-    apply(sh, 0);
+  out_apply_status:
+    tlog_debug("log_read_complete exit with status %v\n", s);
+    buffer_clear(tl->tuple_staging);
+    apply(sh, s);
+  out:
     closure_finish();
 }
 
-void read_log(log tl, u64 offset, u64 size, status_handler sh)
+static boolean init_staging(log tl, status_handler sh)
```

Comment:
does seem like we save a little repetition by just always fixing this up in read_log?
never mind...forgot about the filling from empty case